lib/build.js

```diff
@@ -505,7 +505,7 @@ class BuildModel extends BaseModel {
 
             return this.getSteps().then(steps => {
                 if (steps.length === 0) {
-                    return new Promise();
+                    throw new Error('Steps do not exist');
```

Comment:
This error seems not to be caught in at least L526 and [here](https://github.com/screwdriver-cd/screwdriver/blob/f805359255d903c153d75de39fad0413e253dbe4/plugins/builds/update.js#L119), is it fine?
And I think we should return a correct Promise object, not an error.

Reply:
Thanks, fixed to return Promise.resolve()